src/System.Management.Automation/engine/CommandPathSearch.cs

```diff
@@ -110,7 +110,17 @@ private void ResolveCurrentDirectoryInLookupPaths()
                 sessionState.CurrentDrive.Provider.NameEquals(fileSystemProviderName) &&
                 sessionState.IsProviderLoaded(fileSystemProviderName);
 
-            string environmentCurrentDirectory = Directory.GetCurrentDirectory();
+            string? environmentCurrentDirectory = null;
+            
+            try
+            {
+                environmentCurrentDirectory = Directory.GetCurrentDirectory();
```

Comment:
This can have a side effect - if user intend to run the application from current directory but really we will find an application with the same name in another directory.
I don't think it is critical race condition for users but security group can have another opinion.

Reply:
I just checked and if you reference an exe in the current directly `./hello`, it will fail and doesn't just search for another one in PATH.  So we are good on this one.